Makefile

```diff
@@ -0,0 +1,94 @@
+SPEL_AMIUTILSOURCE ?= https://github.com/ferricoxide/Lx-GetAMI-Utils.git
+SPEL_AWSCLISOURCE ?= https://s3.amazonaws.com/aws-cli/awscli-bundle.zip
+SPEL_EPEL7RELEASE ?= https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+SPEL_CUSTOMREPORPM7 ?= https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+SPEL_EPELREPO ?= epel
+SPEL_EXTRARPMS ?= https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_amd64/amazon-ssm-agent.rpm,python36
+SOURCE_AMI_CENTOS7_HVM ?= ami-090b9dabe1c9f40b3
+SOURCE_AMI_RHEL7_HVM ?= ami-0394fe9914b475c53
+SSH_INTERFACE ?= public_dns
+PIP_URL ?= https://bootstrap.pypa.io/get-pip.py
+PYPI_URL ?= https://pypi.org/simple
+
+.PHONY: all install pre_build build post_build
+.EXPORT_ALL_VARIABLES:
+
+ifeq ("${SPEL_CI}", "true")
+SPEL_VERSION=$(shell date +%Y.%m.dev%s)
+endif
+
+$(info SPEL_IDENTIFIER=${SPEL_IDENTIFIER})
+$(info SPEL_VERSION=${SPEL_VERSION})
+
+ifndef SPEL_IDENTIFIER
+$(error SPEL_IDENTIFIER is not set)
+endif
+
+ifndef SPEL_VERSION
+$(error SPEL_VERSION is not set)
+endif
+
+$(shell mkdir -p ".spel/${SPEL_VERSION}/")
+$(shell touch ".spel/${SPEL_VERSION}/packer.log")
+#PACKER_LOG_PATH=".spel/${SPEL_VERSION}/packer.log"
+
+# Check if $SPEL_PIN_AWSCLI_BUNDLE is not null
+ifdef SPEL_PIN_AWSCLI_BUNDLE
+AWSCLI_PIN=$(shell grep aws-cli requirements/aws-cli.txt)
+ifndef AWSCLI_PIN
+$(error AWSCLI_PIN is not set: ${.SHELLSTATUS})
+endif
+
+AWSCLI_VERSION:=$(subst aws-cli==,,${AWSCLI_PIN})
+SPEL_AWSCLISOURCE := $(shell dirname ${SPEL_AWSCLISOURCE}/awscli-bundle-${AWSCLI_VERSION}.zip)
+ifndef SPEL_AWSCLISOURCE
+$(error SPEL_AWSCLISOURCE is not set: ${SHELLSTATUS})
+endif
+endif
+$(info SPEL_AWSCLISOURCE=${SPEL_AWSCLISOURCE})
+
+# Check if $SPEL_SSM_ACCESS_KEY is not null
+ifdef SPEL_SSM_ACCESS_KEY
+SSM_ACCESS_KEY = $(shell aws ssm get-parameters --name ${SPEL_SSM_ACCESS_KEY} --with-decryption --query 'Parameters[0].Value' --out text)
+$(shell aws configure set aws_access_key_id ${SSM_ACCESS_KEY} --profile ${SPEL_IDENTIFIER})
+
+SSM_SECRET_KEY = $(shell aws ssm get-parameters --name ${SPEL_SSM_SECRET_KEY} --with-decryption --query 'Parameters[0].Value' --out text)
+$(shell aws configure set aws_secret_access_key ${SSM_SECRET_KEY} --profile ${SPEL_IDENTIFIER})
+# Setup the profile credentials
+else ifdef AWS_ACCESS_KEY_ID
+$(shell aws configure set aws_access_key_id ${AWS_ACCESS_KEY_ID} --profile ${SPEL_IDENTIFIER})
+$(shell aws configure set aws_secret_access_key ${AWS_SECRET_ACCESS_KEY} --profile ${SPEL_IDENTIFIER})
+endif 
+
+# Check if $SPEL_SSM_SESSION_TOKEN is not null
+ifdef SPEL_SSM_SESSION_TOKEN
+SSM_SESSION_TOKEN_PARAMS = $(shell aws ssm get-parameters --name $SPEL_SSM_SESSION_TOKEN --with-decryption --query 'Parameters[0].Value' --out text)
+$(shell aws configure set aws_session_token ${SSM_SESSION_TOKEN_PARAMS} --profile ${SPEL_IDENTIFIER})
+endif
+
+$(shell aws configure set region ${AWS_REGION} --profile ${SPEL_IDENTIFIER})
+
+all: install pre_build build post_build
```

Comment:
Something to watch out for with `make` is that it is not necessarily sequential. It will resolve all the targets and their dependencies, and it _may_ run them in any order allowed by the dependencies. That it _typically_ runs them in the listed order is a matter of convention, not its spec. For this `all` target to work in all cases, its dependencies need dependencies... But at that point, the `all` target seems to become extraneous.

```
install:
	bash ./build/install.sh

pre_build: install
	bash ./build/pre_build.sh

build: pre_build
	bash ./build/build.sh
 
post_build:
	bash ./build/post_build.sh